AutoCollection/HttpRequests.ts

```diff
@@ -14,6 +13,7 @@ import AutoCollectPerformance = require("./Performance");
 class AutoCollectHttpRequests {
 
     public static INSTANCE: AutoCollectHttpRequests;
+    public static HANDLER_READY: boolean = false;
```

Comment:
I'm trying to remove overuse of static stuff in this SDK, no need to expose this one, just leave this one as private not static property here

Reply:
Unfortunately we need a static variable to know if the handler is set on the createServer function.

.setup() creates a new instance of AutoCollectHttpRequests even if an instance currently exists. This means that if there is no static variable to say that the handler has been created beforehand a second handler will be added to the createServer function. The problem that will occur, without a static variable to tell if the handler has been created, is if the user makes a call to .setup() then .dispose()+.setup() then .dispose()+.setup(), ... and after call http.createServer. With this scenario, these requests will be recorded several times :/

Reply:
What can be done is to change the behavior of the .setup() function.
- If _serverRequests is null we create an instance
- If _serverRequests is not null we update the _client variable

Do you think it would be better?

Reply:
I see, leaving as static should be fine then, just leave it as private, also looks like http basic usage tests are [failing](https://github.com/microsoft/ApplicationInsights-node.js/runs/6040472929?check_suite_focus=true) in EndToEnd test class because of this change.

Reply:
Yes, I saw it. I'm going to watch it tonight to try to figure out what's going on